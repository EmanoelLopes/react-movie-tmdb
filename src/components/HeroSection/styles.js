import styled from 'styled-components';
import { media, images } from 'utils/styled';

const { path, backdrop, placeholder } = images;

export const Hero = styled.section`
  background-attachment: fixed;
  background-image: url(
    ${props => (props.backdrop) 
      ? `${path}${backdrop}${props.backdrop}`
      : `${placeholder.backdrop}`
    }
  );
  background-position: center center;
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  width: 100%;

  ${media.desktop`
    height: 700px;
    background-position: center 0;
  `}

  .hero-info {
    background-color: rgba(0, 0, 0, .7);
    color: #fff;
    display: flex;
    justify-content: center;
    padding: 50px 0;

    .container {
      ${media.desktop`
        flex-direction: row;
      `}
    }

    h1 {
      font-size: 32px;
      text-align: center;
      text-transform: uppercase;

      ${media.desktop`
        text-align: left;
      `}
    }

    h2 {
      font-size: 28px;
    }

    h3 {
      font-size: 22px;
    }

    h1, h2, h3 {
      margin-bottom: 15px; 
    }

    p {
      font-size: 22px;
    }
  }

  .hero-poster {
    ${media.desktop`
      margin-right: 25px;
    `}


    figure {
      text-align: center;
      margin-bottom: 15px;
    }

    img{
      max-width: 300px;
      border: 3px solid #fff;

      ${media.desktop`
        max-width: 200px;
      `}
    }
  }
`;
