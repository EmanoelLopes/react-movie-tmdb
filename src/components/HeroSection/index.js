import React, { Fragment } from 'react';
import { string, bool, array, number } from 'prop-types';
import { imagesPath } from 'utils/styled';
import { Hero } from './styles';

const HeroSection = props => {
  const {
    title,
    description,
    backdrop,
    isCurrent,
    posterPath,
    releaseDate,
    createdBy,
    rating,
  } = props;

  return (
    <Hero backdrop={backdrop}>
      <div className="hero-info">
        <div className="container">
          {(isCurrent) &&
            <div className="hero-poster">
              <figure>
                <img src={`${imagesPath}/w500${posterPath}`} alt={title} />
              </figure>
            </div>
          }
          <div className="hero-overview">
            <h1>{title}</h1>
            {(isCurrent) && 
              <Fragment>
                <h2>First Release: {releaseDate}</h2>
                <h3>Directors: {createdBy.map((director, index) => {
                    return (
                      <span key={director.id}>      
                        {director.name} 
                        {createdBy.length -1 !== index ? ', ': '' }
                      </span>
                    )
                  })
                }</h3>
                <h3>
                  Rating:
                  <span className="vote-average">
                    <small style={{ width: `${rating/10 * 100}%`}}></small>
                  </span> {rating}
                </h3>
              </Fragment>
            }
            <p>Overview: {description}</p>
          </div>
        </div>
      </div>
    </Hero>
  );
};

HeroSection.propTypes = {
  title: string.isRequired,
  backdrop: string.isRequired,
  description: string.isRequired,
  posterPath: string,
  isCurrent: bool,
  releaseDate: string,
  createdBy: array,
  rating: number,
};

HeroSection.defaultProps = {
  title: '',
  backdrop: '',
  description: '',
  posterPath: '',
  isCurrent: false,
  releaseDate: '',
  createdBy: [],
  rating: null,
};

export default HeroSection;
