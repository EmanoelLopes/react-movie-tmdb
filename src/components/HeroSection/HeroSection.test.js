import React from 'react';
import { mount } from 'enzyme';
import HeroSection from 'components/HeroSection';

let wrapped;
let defaultProps = {
  title: '',
  description: '',
  backdrop: '',
};

beforeEach(() => {
  wrapped = mount(<HeroSection {...defaultProps} />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('<HeroSection />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should have a tag <h1> with a text', () => {
    expect(wrapped.find('.hero-info h1')).toHaveLength(1);
  });

  it('should have a tag <p> with a text', () => {
    expect(wrapped.find('.hero-info p')).toHaveLength(1);
  });
});
