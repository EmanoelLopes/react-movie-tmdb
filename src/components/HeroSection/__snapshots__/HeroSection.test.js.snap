// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroSection /> should match snapshot 1`] = `
.c0 {
  background-attachment: fixed;
  background-image: url(https://image.tmdb.org/t/p/w1280);
  background-position: 0 0;
  background-size: cover;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
}

.c0 .hero-info {
  background-color: rgba(0,0,0,.7);
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 50px 0;
}

.c0 .hero-info h1 {
  font-size: 32px;
  text-align: center;
  text-transform: uppercase;
}

.c0 .hero-info h2 {
  font-size: 28px;
}

.c0 .hero-info h3 {
  font-size: 22px;
}

.c0 .hero-info h1,
.c0 .hero-info h2,
.c0 .hero-info h3 {
  margin-bottom: 15px;
}

.c0 .hero-info p {
  font-size: 22px;
}

.c0 .hero-info .vote-average {
  background-color: #fff;
  display: inline-block;
  height: 25px;
  margin: 0 15px;
  max-width: 200px;
  vertical-align: middle;
  width: 100%;
}

.c0 .hero-info .vote-average small {
  background-color: #01d277;
  display: block;
  height: 100%;
}

.c0 .hero-poster figure {
  text-align: center;
  margin-bottom: 15px;
}

.c0 .hero-poster img {
  max-width: 300px;
  border: 3px solid #fff;
}

@media (min-width:62em) {
  .c0 {
    height: 700px;
  }
}

@media (min-width:62em) {
  .c0 .hero-info .container {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:62em) {
  .c0 .hero-info h1 {
    text-align: left;
  }
}

@media (min-width:62em) {
  .c0 .hero-poster {
    margin-right: 25px;
  }
}

@media (min-width:62em) {
  .c0 .hero-poster img {
    max-width: 200px;
  }
}

<HeroSection
  backdrop=""
  createdBy={Array []}
  description=""
  genres={Array []}
  isCurrent={false}
  posterPath=""
  rating={null}
  releaseDate=""
  title=""
>
  <styled.section
    backdrop=""
  >
    <StyledComponent
      backdrop=""
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  background-attachment: fixed;
  background-image: url(",
              [Function],
              ");
  background-position: 0 0;
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  width: 100%;

  ",
              "
    @media (min-width: ",
              "62",
              "em) {
      ",
              "
    height: 700px;
  ",
              "
    }
  ",
              "

  .hero-info {
    background-color: rgba(0, 0, 0, .7);
    color: #fff;
    display: flex;
    justify-content: center;
    padding: 50px 0;

    .container {
      ",
              "
    @media (min-width: ",
              "62",
              "em) {
      ",
              "
        flex-direction: row;
      ",
              "
    }
  ",
              "
    }

    h1 {
      font-size: 32px;
      text-align: center;
      text-transform: uppercase;

      ",
              "
    @media (min-width: ",
              "62",
              "em) {
      ",
              "
        text-align: left;
      ",
              "
    }
  ",
              "
    }

    h2 {
      font-size: 28px;
    }

    h3 {
      font-size: 22px;
    }

    h1, h2, h3 {
      margin-bottom: 15px; 
    }

    p {
      font-size: 22px;
    }

    .vote-average {
      background-color: #fff;
      display: inline-block;
      height: 25px;
      margin: 0 15px;
      max-width: 200px;
      vertical-align: middle;
      width: 100%;

      small {
        background-color: #01d277;
        display: block;
        height: 100%;
      }
    }
  }

  .hero-poster {
    ",
              "
    @media (min-width: ",
              "62",
              "em) {
      ",
              "
      margin-right: 25px;
    ",
              "
    }
  ",
              "


    figure {
      text-align: center;
      margin-bottom: 15px;
    }

    img{
      max-width: 300px;
      border: 3px solid #fff;

      ",
              "
    @media (min-width: ",
              "62",
              "em) {
      ",
              "
        max-width: 200px;
      ",
              "
    }
  ",
              "
    }
  }
",
            ],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="c0"
      >
        <div
          className="hero-info"
        >
          <div
            className="container"
          >
            <div
              className="hero-overview"
            >
              <h1 />
              <p>
                Overview: 
              </p>
            </div>
          </div>
        </div>
      </section>
    </StyledComponent>
  </styled.section>
</HeroSection>
`;
